metrics/exporter.go

```diff
@@ -187,6 +187,10 @@ func isNewExporterRequired(newConfig *metricsConfig) bool {
 		return newConfig.collectorAddress != cc.collectorAddress || newConfig.requireSecure != cc.requireSecure
 	}
 
+	if newConfig.backendDestination == prometheus {
+		return newConfig.prometheusPort != cc.prometheusPort
+	}
+
 	return newConfig.backendDestination == stackdriver && newConfig.stackdriverClientConfig != cc.stackdriverClientConfig
 }
```

Comment:
return dest=promo && nc.pp != cc.pp || des == stack && nc.sCC != cc.sCC?

Reply:
I might make this a switch, I worry about having a conditional with 4 boolean expressions being hard to maintain.